File: BF_CC_GoLang/bfcc.go
package main

import (
	"fmt"
	"strings"
)

func bruteForce(cipherText string) {
	for i := 1; i <= 25; i++ { // we try all possible shift values from 1 to 25
		plainText := ""
		for _, ch := range cipherText {
			if ch >= 'a' && ch <= 'z' {
				plainText += string('a' + ((ch-'a'+rune(i))%26))
			} else if ch >= 'A' && ch <= 'Z' {
				plainText += string('A' + ((ch-'A'+rune(i))%26))
			} else {
				plainText += string(ch)
			}
		}
		fmt.Printf("Shift %d: %s\n", i, plainText)
	}
}

func main() {
	cipherText := "Frph ryhu khuh Zdwvrq"
	bruteForce(strings.ToLower(cipherText)) // we convert the ciphertext to lowercase
}

File: BFCC_Python/bfcc.py
import string

def brute_force(cipher_text):
    # Create a list of all possible characters
    characters = string.ascii_letters + string.digits + string.punctuation + " "

    # Try all possible combinations of the characters
    for key in range(len(characters)):
        # Decrypt the cipher text using the key
        plain_text = ""
        for char in cipher_text:
            index = characters.find(char)
            index = (index - key) % len(characters)
            plain_text += characters[index]

        # Print the decrypted text and the key used to decrypt it
        print("Key: " + str(key) + " | Decrypted text: " + plain_text)

encrypted = 'Frph ryhu khuh Zdwvrq'
print(encrypted)
brute_force(encrypted)
